runtime/metricsview/executor_rewrite_druid_groups.go

```diff
@@ -17,8 +17,9 @@ func (e *Executor) rewriteDruidGroups(ast *AST) error {
 }
 
 func (e *Executor) rewriteDruidGroupsWalk(n *SelectNode) error {
-	// Skip if already grouped
-	if n.Group {
+	// Skip if already grouped or if there is no sub query with JOIN
+	// Druid requires GROUP BY and ANY_VALUE aggregate for measure if there is a subquery with JOIN
+	if n.Group || (n.SpineSelect == nil && len(n.LeftJoinSelects) == 0 && n.JoinComparisonSelect == nil) {
```

Comment:
And what about `SpineSelect`? That is `RIGHT JOIN`ed to the query

Reply:
Just checked with measure filter, it seems like we may not add group by here however the query is getting converted to a new `windowOperator` query and I am already hitting some edge cases with this query. We may need to look into upgrading druid cluster to get latest version of this query, till then lets not make change for this.